rwengine/include/script/ScriptTypes.hpp

```diff
@@ -66,13 +68,19 @@ struct SCMOpcodeParameter {
 
 	int integerValue() const 
 	{
-		if ( type == TGlobal )
-		{
-			return * globalInteger;
-		}
-		else
+		switch (type)
 		{
-			return integer;
+		case TGlobal:
+			return *globalInteger;
+		case TInt8:
+			return (uint8_t)integer;
+		case TInt16:
+			return (uint16_t)integer;
+		case TInt32:
+			return (uint32_t)integer;
```

Comment:
Yeah, these should be signed.
On 20 Jun 2016 2:03 am, "Jannik Vogel" <notifications@github.com> wrote:

> In rwengine/include/script/ScriptTypes.hpp
> <https://github.com/rwengine/openrw/pull/147#discussion_r67627419>:
>
> >  		{
> > -			return integer;
> > +		case TGlobal:
> > +			return *globalInteger;
> > +		case TInt8:
> > +			return (uint8_t)integer;
> > +		case TInt16:
> > +			return (uint16_t)integer;
> > +		case TInt32:
> > +			return (uint32_t)integer;
>
> Looks like I changed these to signed ints in my commit
> <https://github.com/JayFoxRox/openrw/pull/11/commits/5455d2912675419032292442732c6c759813b393>
>
> I can't remember where it mattered specifically, but I probably did it for
> a reason.
> I didn't see any forced signed comparisions in SCM yet either so it's
> probably signed by default.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rwengine/openrw/pull/147/files/dfcfdd0cb84e6a3aef3c98fbf13711dbacc1c129#r67627419>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAZho61lCoPE47gGAa-aDLzCqtxhBAsLks5qNeb4gaJpZM4I5D4g>
> .
>